internal/controller/standalone_pgadmin/postgrescluster.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Crunchy Data Solutions, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package standalone_pgadmin
+
+import (
+	"context"
+
+	"github.com/crunchydata/postgres-operator/internal/naming"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//+kubebuilder:rbac:groups="postgres-operator.crunchydata.com",resources="pgpgadmins",verbs={list}
+
+// findPGAdminsForPostgresCluster returns PGAdmins that target a given cluster.
+func (r *PGAdminReconciler) findPGAdminsForPostgresCluster(
+	ctx context.Context, cluster client.Object,
+) []*v1beta1.PGAdmin {
+	var (
+		matching []*v1beta1.PGAdmin
+		pgadmins v1beta1.PGAdminList
+	)
+
+	// NOTE: If this becomes slow due to a large number of pgadmins in a single
+	// namespace, we can configure the [ctrl.Manager] field indexer and pass a
+	// [fields.Selector] here.
+	// - https://book.kubebuilder.io/reference/watching-resources/externally-managed.html
+	if r.List(ctx, &pgadmins, &client.ListOptions{
+		Namespace: cluster.GetNamespace(),
+	}) == nil {
+		for i := range pgadmins.Items {
+			for _, serverGroup := range pgadmins.Items[i].Spec.ServerGroups {
+				if selector, err := naming.AsSelector(*serverGroup.PostgresClusterSelector); err == nil {
```

Comment:
⛏️ There's a dereference here that doesn't check for nil first.

I see that the field is required in the CRD. We can reflect that in the structs by making it a value instead of a pointer.

Reply:
(Oof, that's a pointer because it's a remnant of a test I ran, thx!)